src/libraries/Common/src/System/CodeDom/CodeTypeReference.cs

```diff
@@ -303,7 +303,7 @@ public CodeTypeReference(CodeTypeReference arrayType, int rank)
             ArrayElementType = arrayType;
         }
 
-        public CodeTypeReference ArrayElementType { get; set; }
+        public CodeTypeReference? ArrayElementType { get; set; }
 
         public int ArrayRank { get; set; }
 
```

Comment:
On line 312: BaseType should have [AllowNull] attribute.

Reply:
Similar comment as https://github.com/dotnet/runtime/pull/41476#discussion_r479415576. We never need to set it to `null` in this assembly. When we tackle System.CodeDom we will need to publicly annotate this type.